lmnet/lmnet/networks/classification/lm_resnet.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import functools
+import tensorflow as tf
+
+from lmnet.networks.classification.base import Base
+from lmnet.layers import fully_connected
+
+
+class LmResnet(Base):
+    """Residual network (ResNet) of 18-layers for classification
+
+    This ResNet-18 is modified from the ImageNet version of ResNet-18 of the original paper
+            Deep Residual Learning for Image Recognition (https://arxiv.org/abs/1512.03385)
+
+    - The first layer is 3x3 convolution layer with stride 1 instead of 7x7 conv with stride 2,
+      like the CIFAR-10 version of ResNet-18 in the paper.
+    - The 3x3 max pooling with stride 2 is not used in this architecture.
+    - In each residual block, batch normalization (BN) is after the add, to be specific, a
+      pre-activation variant of residual block is used.
+    - Utilizing Space-to-Depth operator for each the transition layer, convolution op with
+      strides of 2 is replaced with space-to-depth.
+    - Note currently this ResNet-18 only supports NHWC data format.
+    """
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        assert self.data_format == 'NHWC'
+        self.custom_getter = None
+        self.activation = tf.nn.relu
+        self.weight_decay_rate = 0.0001
```

Comment:
@nlpng
I found little things 😭
Could you change to enable to set `weight_decay_rate` in training config file?